internal/kms/secretskms.go

```diff
@@ -184,7 +165,46 @@ func (kms secretsMetadataKMS) fetchEncryptionPassphrase(
 
 // Destroy frees all used resources.
 func (kms secretsMetadataKMS) Destroy() {
-	kms.secretsKMS.Destroy()
+	// nothing to do
+}
+
+// FetchDEK returns passphrase from Kubernetes secrets.
+func (kms secretsMetadataKMS) FetchDEK(ctx context.Context, key string) (string, error) {
+	var (
+		smKMS                secretsMetadataKMS
+		encryptionPassphrase string
+		ok                   bool
+		err                  error
+	)
+
+	encryptionPassphrase, err = smKMS.fetchEncryptionPassphrase(
+		kms.Config, kms.Tenant)
+	if err != nil {
+		if !errors.Is(err, errConfigOptionMissing) {
+			return "", err
+		}
+		// if 'userSecret' option is not specified, fetch encryptionPassphrase
+		// from storageclass secrets.
+		encryptionPassphrase, ok = kms.Secrets[encryptionPassphraseKey]
```

Comment:
```suggestion
	encryptionPassphrase, err := kms.fetchEncryptionPassphrase(
		kms.Config, kms.Tenant)
	if err != nil {
		if !errors.Is(err, errConfigOptionMissing) {
			return "", err
		}
		// if 'userSecret' option is not specified, fetch encryptionPassphrase
		// from storageclass secrets.
		encryptionPassphrase, ok := kms.Secrets[encryptionPassphraseKey]
```

Reply:
Yes, you are right, i removed the local variable `smKMS`.